src/hooks/useJoinSponsorship.ts

```diff
@@ -0,0 +1,53 @@
+import { useMemo } from 'react'
+import { toaster } from 'toasterhea'
+import { useWalletAccount } from '~/shared/stores/wallet'
+import { useMyOperator } from '~/hooks/useMyOperator'
+import JoinSponsorshipModal from '~/modals/JoinSponsorshipModal'
+import { Layer } from '~/utils/Layer'
+import getSponsorshipTokenInfo from '~/getters/getSponsorshipTokenInfo'
+import { stakeOnSponsorship } from '~/services/sponsorships'
+import { toBN } from '~/utils/bn'
+
+const joinSponsorshipModal = toaster(JoinSponsorshipModal, Layer.Modal)
+
+export const useJoinSponsorship = (): {
+    canJoinSponsorship: boolean
+    joinSponsorship: (sponsorshipId: string, sponsorshipStreamId: string) => Promise<void>
+} => {
+    const wallet = useWalletAccount()
+    const myOperatorQuery = useMyOperator(wallet || '')
+    const canJoinSponsorship = useMemo(() => !!myOperatorQuery.data, [myOperatorQuery])
+
+    return {
+        canJoinSponsorship,
+        joinSponsorship: async (sponsorshipId, sponsorshipStreamId) => {
+            if (!myOperatorQuery.data) {
+                return
+            }
+            const balance = myOperatorQuery.data?.poolValue.minus(
+                myOperatorQuery.data?.stakes.reduce((prev, curr, index) => {
+                    return prev.plus(curr.amount)
+                }, toBN(0)),
+            )
+            try {
```

Comment:
Remove try/catch to make possible errors visible

Reply:
The try/catch here is for the case of closing the modal, which is a promise rejection. As for the possible errors, they are being handled and displayed :)